actors/builtin/miner/miner_actor.go

```diff
@@ -142,66 +140,37 @@ func (a Actor) ChangeWorkerAddress(rt Runtime, params *ChangeWorkerAddressParams
 }
 
 //////////////////
-// SurprisePoSt //
+// FallbackPoSt //
 //////////////////
 
-// Called by Actor to notify StorageMiner of SurprisePoSt Challenge.
-func (a Actor) OnSurprisePoStChallenge(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
-	rt.ValidateImmediateCallerIs(builtin.StoragePowerActorAddr)
-
-	var st State
-	challenged := rt.State().Transaction(&st, func() interface{} {
-		// If already challenged, do not challenge again.
-		// Failed PoSt will automatically reset the state to not-challenged.
-		if st.PoStState.isChallenged() {
-			return false
-		}
-
-		st.PoStState = PoStState{
-			LastSuccessfulPoSt:     st.PoStState.LastSuccessfulPoSt,
-			SurpriseChallengeEpoch: rt.CurrEpoch(),
-			NumConsecutiveFailures: st.PoStState.NumConsecutiveFailures,
-		}
-		return true
-	}).(bool)
-
-	if challenged {
-		// Request deferred Cron check for SurprisePoSt challenge expiry.
-		cronPayload := CronEventPayload{
-			EventType: CronEventType_Miner_SurpriseExpiration,
-			Sectors:   nil,
-		}
-		surpriseDuration := power.SurprisePostChallengeDuration
-		a.enrollCronEvent(rt, rt.CurrEpoch()+surpriseDuration, &cronPayload)
-	}
-	return &adt.EmptyValue{}
-}
-
-type SubmitSurprisePoStResponseParams struct {
-	onChainInfo abi.OnChainSurprisePoStVerifyInfo
-}
+// TODO: where should this live?
+const ProvingPeriod = 300
 
-// Invoked by miner's worker address to submit a response to a pending SurprisePoSt challenge.
-func (a Actor) SubmitSurprisePoStResponse(rt Runtime, params *SubmitSurprisePoStResponseParams) *adt.EmptyValue {
+// Invoked by miner's worker address to submit their fallback post
+func (a Actor) SubmitFallbackPoSt(rt Runtime, params *abi.OnChainPoStVerifyInfo) *adt.EmptyValue {
 	var st State
 	rt.State().Transaction(&st, func() interface{} {
 		rt.ValidateImmediateCallerIs(st.Info.Worker)
-		if !st.PoStState.isChallenged() {
-			rt.Abort(exitcode.ErrIllegalState, "Not currently challenged")
+
+		if rt.CurrEpoch() <= st.PoStState.ProvingPeriodStart+ProvingPeriod {
+			rt.Abort(exitcode.ErrIllegalState, "Not currently in submission window for PoSt")
 		}
-		a.verifySurprisePost(rt, &st, &params.onChainInfo)
+		a.verifyWindowedPost(rt, &st, params)
+
+		// TODO: post is valid, do we do anything with faults now?
```

Comment:
There are two kinds of faults, Miner-Wide fault (failing fallback) and Sector-Specific fault (temporary fault). If a miner is not previously in Miner-Wide Fault (known as DetectedFault previously), then submitting FallbackPoSt should have no impact on sector state wrt fault. If a miner is in Miner-Wide Fault, then submitting FallbackPoSt successfully will get the miner out of the Miner-Wide Fault and put its sectors back in the sectorset. Miners may only gain power after the next PoSt.

Reply:
TemporaryFault recovery on the other hand is independent of Fallback as it has its own EffectiveEnd where the sector is added back to the sector set. To be consistent, it should only regain its power after the first PoSt.

Reply:
Added a call to power actor below